scipy/linalg/tests/test_lapack.py

```diff
@@ -710,3 +730,73 @@ def test_hetrd(self):
             assert_allclose(
                 QHAQ, T, atol=10*np.finfo(real_dtype).eps, rtol=1.0
                 )
+
+
+def test_gglse():
+    # Example data taken from NAG manual
+    for ind, dtype in enumerate(DTYPES):
+
+        func, func_lwork = get_lapack_funcs(('gglse', 'gglse_lwork'),
+                                            dtype=dtype)
+        lwork = _compute_lwork(func_lwork, m=6, n=4, p=2)
+        if ind < 2:
```

Comment:
This would be clearer as ``if dtype in REAL_DTYPES``.

Reply:
Same comment for the identical check below. And then you don't need the ``enumerate``, just ``for dtype in DTYPES``.

Reply:
Yes, as below, we again have 4 routines <s,d> and <c,z> groupings. And for first two we have a float example, and similarly a complex example for the other group and DTYPES goes like single, double, single complex, double complex